core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/RequestHandlerManager.java

```diff
@@ -134,6 +134,21 @@ public boolean isRegistered(String method) {
         return methodToCategory.containsKey(method);
     }
 
+    public synchronized boolean deregister(String method) {
```

Comment:
As I see we can handle this more accurate here: we can first get category by method's name from `methodToCategory` and than knowing category we can remove method handler from exactly one map of handlers, what do you think?